web/themes/custom/pineapple/pineapple.theme

```diff
@@ -54,6 +54,19 @@ function pineapple_preprocess_node(array &$variables) {
   if ($node->getType() == 'recipe') {
     $owner = $node->getOwner();
     $variables['owner'] = $owner;
+
+    if ( isset($node->_referringItem)) {
+      $parent = $node->_referringItem->getEntity()->getType();
+      $variables['parent'] = $parent;
+    }
+  }
+
+  if ($node->getType() == 'article') {
+
+    if ( isset($node->_referringItem)) {
+      $parent = $node->_referringItem->getEntity()->getType();
+      $variables['parent'] = $parent;
+    }
```

Comment:
Somethings to note here 

- code duplication here which isn't good. Do something like ->> if the node type is equal to recipe or article { do parent stuff }
- remove the space before isset 
- rename "parent" to parent entity type because that's what parent actually is (twig will also need updating)

Reply:
done